torch/distributions/constraints.py

```diff
@@ -488,11 +490,21 @@ def check(self, value):
         return torch.isclose(value, value.mT, atol=1e-6).all(-2).all(-1)
 
 
+class _PositiveSemidefinite(_Symmetric):
+    """
+    Constrain to positive-semidefinite matrices.
+    """
+    def check(self, value):
+        sym_check = super().check(value)
+        if not sym_check.all():
+            return sym_check
+        return torch.linalg.eigvalsh(value).ge(0).all(-1)
```

Comment:
For consistency, could it use `all()` in both checks?

Reply:
I don't understand this part. Where should you recommend to use `all()` additionally?
Currently, `sym_check` is a tensor reduced last two dimension of the input `value` tensor.
Also, return object is a tensor reducing two dimension of input `value` tensor since `torch.linalg.eigvalsh` reduces 1 dimension and `.all(-1)` reduces one more dimension.

Reply:
@vadimkantorov: The issue is that we can have batched matrices e.g. of shape (b, n, n) where we expect this function to return a tensor of shape (b) (i.e. the check is run separately for each matrix in the batch), using `all` will end up reducing the batch dimensions as well.

Reply:
I guess it will be fine.